site/docs/transform/impute.md

```diff
@@ -0,0 +1,123 @@
+The values are then grouped by the field `"c"` of the `color` encoding.
+For eac, the impute transform then determines missing key values within each group.
+In this case, the data tuple where `"x"` is `3` and `"c"` is `1` is missing, so a new tuple with `"x"` = `3`, `"c"` = `1`, and `"y"` = `0` will be added.
+
+{:#encoding-impute-value}
+<div class="vl-example" data-name="line_impute_value"></div>
+
+Besides imputing with a constant `value`, we can also use a `method` (such as `"mean"`) on existing data points to generate the missing data.
+
+<div class="vl-example" data-name="line_impute_method"></div>
+
+The `frame` property of `impute` can be used to control the window over which the specified `method` is applied.
+Here, the `frame` is `[-2, 2]` which indicates that the window for calculating mean includes two objects preceding and two objects following the current object.
+
+<div class="vl-example" data-name="line_impute_frame"></div>
+
+### Specifying the Key Values to be Imputed
+
+The `keyvals` property provides additional key values that should be considered for imputation.  If not provided, all of the values will be derived from all unique values of the `key` field.
+If there is no grouping field (e.g., no `color` in the examples given above), then `keyvals` _must_ be specified.  Otherwise, the impute transform will have no effect on the data.
+
+
+The `keyvals` property can be an array:
+
+<div class="vl-example" data-name="line_encoding_impute_keyvals"></div>
+
+
+
+{:#sequence-def}
+
+Alternatively, the `keyvals` property can be an [object](#sequence-def)  defining a sequence, which can contain the following properties:
+
+{% include table.html props="start,stop,step" source="ImputeSequence" %}
+
+<div class="vl-example" data-name="line_encoding_impute_keyvals_sequence"></div>
+
+{:#transform}
+## Impute Transform
+
+An impute transform can also be specified as a part of the `transform` array.
+
+{: .suppress-error}
+```json
+// A View Specification
+{
+  ...
+  "transform": [
+    ...
+    {
+      // Impute Transform
+      "impute": ...,
+      "key": ...,
+      "keyvals": ...,
+      "groupby": [...]
+    }
+    ...
+  ],
+  ...
+}
+```
+
+{% include table.html props="impute,key,keyvals,groupby" source="ImputeTransform" %}
+
+For example, the same chart with `impute` in encoding [above]("#encoding-impute-value") can be created using the `impute` transform.
+Here, we have to manually specify the `key` and `groupby` fields, which were inferred automatically for `impute` in `encoding`.
+
+<div class="vl-example" data-name="line_impute_transform_value"></div>
+
+<div class="vl-example" data-name="line_impute_transform_frame"></div>
+
+Similarly if the `groupby` property is not specified, `keyvals` _must_ be specified.
+
+<div class="vl-example" data-name="line_impute_keyvals"></div>
+
+<div class="vl-example" data-name="line_impute_keyvals_sequence"></div>
```

Comment:
This second example is unnecessary, please remove it from the doc and the example folder.